plugins/woocommerce/src/Blocks/BlockTypes/AddToCartWithOptionsVariationSelectorItemTemplate.php

```diff
@@ -23,4 +24,99 @@ class AddToCartWithOptionsVariationSelectorItemTemplate extends AbstractBlock {
 	protected function get_block_type_style() {
 		return null;
 	}
+
+	/**
+	 * Render the block.
+	 *
+	 * @param array    $attributes Block attributes.
+	 * @param string   $content Block content.
+	 * @param WP_Block $block Block instance.
+	 * @return string Rendered block output.
+	 */
+	protected function render( $attributes, $content, $block ): string {
+		global $product;
+
+		$content = '';
+
+		$product_attributes = $product->get_variation_attributes();
+
+		foreach ( $product_attributes as $product_attribute_name => $product_attribute_terms ) {
+			$content .= $this->get_product_row( $product_attribute_name, $product_attribute_terms, $attributes, $block );
+		}
+
+		return $content;
+	}
+
+	/**
+	 * Get product row HTML.
+	 *
+	 * @param string   $product_attribute_name Product Attribute Name.
+	 * @param array    $product_attribute_terms Product Attribute Terms.
+	 * @param array    $attributes Block attributes.
+	 * @param WP_Block $block The Block.
+	 * @return string Row HTML
+	 */
+	private function get_product_row( $product_attribute_name, $product_attribute_terms, $attributes, $block ): string {
+		global $attribute_name;
+		global $attribute_terms;
+
+		$attribute_name = $product_attribute_name;
+		$attribute_terms = $this->get_terms( $product_attribute_name, $product_attribute_terms );
```

Comment:
What are the pros and cons of introducing these globals? I wonder if we could use block context instead.

Reply:
done